From b4bddcb09e5ecad5ce065ec3a9d0064404a8ae6b Mon Sep 17 00:00:00 2001
From: Ningxin Hu <ningxin@intel.com>
Date: Wed, 31 Aug 2016 14:22:08 +0000
Subject: [PATCH 3/3] [ostro-os-xt] Disable namespace and seccomp-bpf sandbox
 on Ostro OS XT.

The namespace sandbox causes renderer fails to send ping message to browser.
The seccomp-bpf sandbox causes renderer crashes on syssig handler.
---
 xwalk/runtime/browser/xwalk_browser_main_parts.cc | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/xwalk/runtime/browser/xwalk_browser_main_parts.cc b/xwalk/runtime/browser/xwalk_browser_main_parts.cc
index 0ee53a9..9c5ae5c 100644
--- a/xwalk/runtime/browser/xwalk_browser_main_parts.cc
+++ b/xwalk/runtime/browser/xwalk_browser_main_parts.cc
@@ -108,6 +108,16 @@ XWalkBrowserMainParts::XWalkBrowserMainParts(
   // content_switches.h.
   base::CommandLine::ForCurrentProcess()->AppendSwitch(
       "disable-setuid-sandbox");
+
+  // FIXME: disable the namespace and seccomp-bpf sandbox on Ostro OS XT.
+  // The namespace sandbox causes renderer fails to send ping message
+  // to browser.
+  base::CommandLine::ForCurrentProcess()->AppendSwitch(
+      "disable-namespace-sandbox");
+  // The seccomp-bpf sandbox causes renderer crashes on syssig handler.
+  base::CommandLine::ForCurrentProcess()->AppendSwitch(
+      "disable-seccomp-filter-sandbox");
+
 #endif
 }
 
-- 
2.8.4

